### PR TITLE
6 ticket2a get request developer only

### LINKED ISSUE #6: Ticket2A - GET REQUEST - Developer only
| name  | about | title  | labels | assignees  |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| "👥 User story"  | Create a new user story  | LINK API  | GET REQUEST | “Gayle  | 

## Story: GET REQUEST (developer only) From Link API

- *As a ….developer,

- **I want to….. link the GitHub API to the backend

- *So That ….. I can pull data from the different GitHub repos.

## Acceptance Criteria

1.GIVEN ... I want a user to see the GitHub PR contributions of any trainee
- WHEN ...I click on the trainees GitHub Link
- THEN ...I should be able to retrieve all the trainee’s GitHub PR’s from the linked GitHub API

2. Given .............
   - When .............
   - Then .............

## Implementation Notes

1.
2.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by Gayle-Thompson-Igwebike**
#6-GET-REQUEST-Gayle-Thompson-Igwebike

## GET REQUEST 

Created a GET request and tested it. Updated the request to fetch all team members PRs 

#6 

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #6 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: server/getPRCount.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const repoName = "Affirmation-page";
+
+// API URL to fetch pull request data from GitHub
+const apiUrl = `https://api.github.com/repos/${repoOwner}/${repoName}/pulls?state=all`;
+
+axios
+  .get(apiUrl)
+  .then((response) => {
+    const pullRequestCount = response.data.length;
+    console.log(`Total Number of pull request contributions from ${repoOwner} in this repo is: ${pullRequestCount}`);
```
Comment (AndrianaOS) [inline server/getPRCount.js]: instead of console logging the response, return it as a res.send/res.json

File: server/getPRCount.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  })
+  .catch((error) => {
+    console.error("Error fetching pull requests:", error);
```
Comment (AndrianaOS) [inline server/getPRCount.js]: Place a status code in the catch so if it catches an error it doesn't still have the status as 200. 
So something like: 
`res.status(500).json({error: error})`

File: server/getPRCount.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const repoOwner = "Gayle-Thompson-Igwebike";
+const repoName = "Affirmation-page";
+
+// API URL to fetch pull request data from GitHub
+const apiUrl = `https://api.github.com/repos/${repoOwner}/${repoName}/pulls?state=all`;
+
+axios
+  .get(apiUrl)
+  .then((response) => {
+    const pullRequestCount = response.data.length;
```
Comment (AndrianaOS) [inline server/getPRCount.js]: Put an if statement in there so if theres an error, it throws the error message to the catch block
`if(!response.data) {res.send("Some error message")}
else {res.status(200).json({message: "Some success string"})}`

File: server/index.js [hunk 4]
```diff
@@ -1,24 +1,28 @@
+import dotenv from "dotenv";
+dotenv.config();
 import express from "express";
 import pkg from "pg";
-
+const axios = require("axios");
+const app = express();
+const port = 8000;
```
Comment (AndrianaOS) [inline server/index.js]: Push the rest of the code we did this afternoon to this branch

**Review by AndrianaOS — CHANGES_REQUESTED** [review]
Great code Gayle, just made a few suggestions to some of the responses.
Remember to push your code from this afternoon in as well

**Comment by AndrianaOS** [comment]
Also remember to pull before you push changes to resolve the merge conflict in this branch

**Comment by Gayle-Thompson-Igwebike** [comment]
closed this branch as the get request was made on another branch - get-request

**Pull request closed without merging**